src/lib.rs

```diff
@@ -156,7 +168,7 @@ where
     }
 }
 
-enum DocumentAttributeParserState<'a> {
+pub enum DocumentAttributeParserState<'a> {
```

Comment:
If this is pub, it should be documented. But, I don't think this should be exposed on the public API.

Reply:
I'd like to be able to tell when it's done, it isn't required but it's kinda nice to have